server/db/mysql/adapter.go

```diff
@@ -840,7 +840,9 @@ func (a *adapter) UserGetAll(ids ...t.Uid) ([]t.User, error) {
 
 		users = append(users, user)
 	}
-	rows.Close()
+	if err := rows.Err(); err != nil {
```

Comment:
Likewise, there is no value in allocating `err` here when it's already allocated. Please use `err =` instead of `err :=`.

Reply:
I would write it differently:
```go
for rows.Next() {
	if err = rows.StructScan(&top); err != nil {
		break
	}
.....
}
if err == nil {
	err = rows.Err()
}
row.Close()
```
It would maintain the flow with a single return and would require no `defer`.

Reply:
Overwriting err and breaking out of loop is not as straight forward as returning early when err is encountered.
`if err != nil { return err }` keeps the code simple and ensures you don't proceed further with spoiled results.